quickfixj-core/src/main/java/quickfix/Session.java

```diff
@@ -584,7 +584,7 @@ public String getRemoteAddress() {
     private boolean isCurrentSession(final long time)
             throws IOException {
         return sessionSchedule == null || sessionSchedule.isSameSession(
-                SystemTime.getUtcCalendar(time), SystemTime.getUtcCalendar(state.getCreationTime()));
+                SystemTime.getUtcCalendar(time), state.getCreationTimeCalendar());
```

Comment:
This avoids creating a calendar from a date-time that's itself created from a calendar (which is the behaviour of all the implementations of `getCreationTime()`).